docs/docs/anatomy.md

```diff
@@ -0,0 +1,254 @@
+---
+sidebar_position: 3
+---
+
+# Anatomy of an Equinox Service
+
+An Equinox service is composed of a few modules (namespaces).
+
+- **Stream**: tells us which category the service writes to and how the identity
+  of the stream is composed (and the reverse operations for when running Reaction logic)
+- **Events**: tells us which events the service writes
+- **Fold**: tells us how the events are folded into state
+- **Decide**: tells us which actions can be taken upon the state (resulting in
+  new events being written)
+- **Service**: the class that wraps the above into a cohesive domain service.
+- **Config**: tells us how to wire the service against concrete stores,
+  including important information like access strategies.
+
+Let's see what each of these modules looks like for a simple service for
+checking in and out of an appointment.
+
+## The Stream
+
+```ts
+export namespace Stream {
+  export const category = "AppointmentActuals"
+  export const streamId = StreamId.gen(AppointmentId.toString, UserId.toString)
+  export const decodeId = StreamId.dec(AppointmentId.parse, UserId.parse)
+  export const tryMatch = StreamName.tryMatch(category, decodeId)
+}
+```
+
+## The Events
+
+```ts
+export namespace Events {
+  // prettier-ignore
+  const date = z.string().datetime().transform((x) => new Date(x))
+  export const Timestamp = z.object({ timestamp: date })
+  export const ActualsOverridden = z.object({ checkedIn: date, checkedOut: date })
+  export type ActualsOverridden = z.infer<typeof ActualsOverridden>
+
+  export type Event =
+    | { type: "CheckedIn"; data: Timestamp }
+    | { type: "CheckedOut"; data: Timestamp }
+    | { type: "ActualsOverridden"; data: ActualsOverridden }
+
+  export const codec = Codec.upcast<Event>(
+    Codec.json(),
+    Codec.Upcast.from({
+      CheckedIn: Timstamp.parse,
+      CheckedOut: Timstamp.parse,
+      ActualsOverridden: ActualsOverridden.parse,
+    }),
+  )
+}
+```
+
+## The Fold
+
+```ts
+export namespace Fold {
+  import Event = Events.Event
+  export type State = { checkedIn?: Date; checkedOut?: Date }
+  export const initial: State = {}
+  export const evolve = (state: State, event: Event) => {
+    switch (event.type) {
+      case "CheckedIn":
+        return { ...state, checkedIn: event.data.timestamp }
+      case "CheckedOut":
+        return { ...state, checkedOut: event.data.timestamp }
+      case "ActualsOverridden":
+        return event.data
+    }
+  }
+}
+```
+
+## The Decide
+
+```ts
+export namespace Decide {
+  import Event = Events.Event
+  import State = Fold.State
+  export const checkIn =
+    (timestamp: Date) =>
+    (state: State): Event[] => {
+      if (state.checkedIn) {
```

Comment:
while this is subjective and has pros
I think 
1. doing a check on the conjunction and throwing
2. then having an if state.checkedIn vs not to illustrate silent idempotent handling as the main thing
3. then the happy path
might convey the rules of the game better (also `no change` conveyed it to me, but its a bit subtle and might merit something more long winded like saying 'caller is expected to guard against this' and `we accept retried requests (with identical timestamps)`)